### ISSUE TITLE
How to render h1 when typing in `# `

### ISSUE THREAD

**Issue by lbhfuture** [visual-space/visual-editor#223]
I want to implement WYSIWYG with markdown syntax, such as auto rendering h1 when typing in `# `(# and followed by a whitespace). I cannot found somewhere to tell me how to do it, could you tell me some tips or demo codes? Thanks.

**Comment by adrian-moisa** [comment]
There are 2 ways of doing it: 
- Use a md editor which can render md text as rich text. Most of them I checked an year ago had separate view for input and text preview.
- Use an editor that has support for typing shortcuts. These typing shortcuts can be used to trigger the insertion of equivalent text elements when md notation is typed in. That does not mean that the document is stored as md. For VE that means it's still stored as delta doc. Super editor can save md. I don't know if they have support for typing shortcuts. Probably they have. As for typing shortcuts for VE, they are not yet implemented. They are planned for #172. We plan to introduce several such behaviours including some of the most used md notations. However that does not mean we support md or we can have typing shortcuts for the entire md syntax.
- A third way would be to use convertors from md to delta and back again. I think there was some work done by some guy but I don't recall the repo name. Overall this approach is rather complex and can't support live changes from md to rich text in the same input element. Only typing shortcuts can do this.

tldr: for now no support in VE but will be added later, ETA end of year. Unless you want to tackle the problem yourself and speed up the process. We are stuck with many other duties. VE is just a small part of a bigger project we run and some VE tasks have been delayed. That's why we did not even publish it yet on pub dev.

**Comment by lbhfuture** [comment]
Thanks for your replying and your works in VE.
